src/components/i18n/DopplerIntlProvider.test.tsx

```diff
@@ -1,33 +1,97 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
 import { DopplerIntlProvider } from "./DopplerIntlProvider";
-import { render } from "@testing-library/react";
-import "@testing-library/jest-dom/extend-expect";
 import { FormattedMessage } from "react-intl";
+import { useAppSessionState } from "../AppSessionStateContext";
+import { AppSessionState } from "../../abstractions/app-session";
 
-describe("DopplerIntlProvider", () => {
-  it("should render Campaign in English", () => {
-    const { getByText } = render(
-      <DopplerIntlProvider locale="en">
-        <FormattedMessage id="campaign" />
-      </DopplerIntlProvider>
-    );
-    getByText("Campaign");
-  });
+jest.mock("../AppSessionStateContext");
+jest.mock("./en", () => ({
+  messages_en: {
+    lang: "en",
+  },
+}));
+jest.mock("./es", () => ({
+  messages_es: {
+    lang: "es",
+  },
+}));
 
-  it("should render Campaign in Spanish", () => {
-    const { getByText } = render(
-      <DopplerIntlProvider locale="es">
-        <FormattedMessage id="campaign" />
-      </DopplerIntlProvider>
+const UNKNOWN_SESSION: AppSessionState = {
+  status: "unknown",
+};
+const NON_AUTHENTICATED: AppSessionState = {
+  status: "non-authenticated",
+};
+const AUTHENTICATED: AppSessionState = {
+  lang: "en",
+  status: "authenticated",
+  jwtToken: "jwt",
+  unlayerUser: {
+    id: "000",
+    signature: "000",
+  },
+  dopplerAccountName: "doppler_mock@mail.com",
+};
+const AUTHENTICATED_WITHOUT_LANG: any | AppSessionState = {
+  status: "authenticated",
+  jwtToken: "jwt",
+  unlayerUser: {
+    id: "000",
+    signature: "000",
+  },
+  dopplerAccountName: "doppler_mock@mail.com",
+};
+const LANG_DEFAULT = "es";
+
+describe(DopplerIntlProvider.name, () => {
+  it.each([
+    { sessionState: UNKNOWN_SESSION },
+    { sessionState: NON_AUTHENTICATED },
+  ])(
+    `should be translate using the default spanish language when user state is $sessionState.status`,
+    ({ sessionState }) => {
+      (useAppSessionState as jest.Mock).mockImplementation(() => sessionState);
+
+      render(
+        <DopplerIntlProvider>
+          <FormattedMessage id="lang" />
+        </DopplerIntlProvider>
+      );
+      screen.getByText(LANG_DEFAULT);
+    }
+  );
+
+  it.each([
+    { userLanguage: "es", language: "spanish" },
+    { userLanguage: "en", language: "english" },
+  ])(
+    `should be translate to $language when user lang exist`,
```

Comment:
```suggestion
    `should translate message to $language when user lang exists`,
```

_DopplerIntlProvider should translate message to $language when user lang exists_